### ISSUE TITLE
Class & Function Selector Does Not Work

### ISSUE THREAD

**Issue by mrclary** [spyder-ide/mac-application#5]
The class and function selector in the editor is visible but the lists only display `<None>`.

This works under bootstrap and build/install.

**Comment by mrclary** [comment]
Okay, I've discovered the issue here. This only occurs under a specific circumstance: upon startup of the standalone mac app if the console python executable is a different version than the spyder python version. Examples: Spyder built with 3.7.7 and conda console is 3.8.2; Spyder built with 3.8.2 and conda console is 3.7.7.

Attached is 
[server_python_21611.log](https://github.com/spyder-ide/mac-application/files/4441241/server_python_21611.log), in which the following error occurs repeatedly:
<details><summary>Error</summary>

```
2020-04-06 16:06:45,369 UTC - ERROR - pyls_jsonrpc.endpoint - Failed to handle request 15
Traceback (most recent call last):
  File "/Users/rclary/Documents/Python.git/developer/spyder-mac-app/dist/Spyder.app/Contents/Resources/lib/python3.8/jedi/inference/compiled/subprocess/__init__.py", line 251, in _send
    is_exception, traceback, result = pickle_load(self._get_process().stdout)
  File "/Users/rclary/Documents/Python.git/developer/spyder-mac-app/dist/Spyder.app/Contents/Resources/lib/python3.8/jedi/_compatibility.py", line 499, in pickle_load
    return pickle.load(file, encoding='bytes')
EOFError: Ran out of input

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/rclary/Documents/Python.git/developer/spyder-mac-app/dist/Spyder.app/Contents/Resources/lib/python3.8/jedi/api/environment.py", line 75, in _get_subprocess
    info = self._subprocess._send(None, _get_info)
  File "/Users/rclary/Documents/Python.git/developer/spyder-mac-app/dist/Spyder.app/Contents/Resources/lib/python3.8/jedi/inference/compiled/subprocess/__init__.py", line 259, in _send
    raise InternalError(
jedi.api.exceptions.InternalError: The subprocess /Users/rclary/anaconda3/envs/spy37-15/bin/python3.7 has crashed (EOFError('Ran out of input'), stderr=).

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "pyls_jsonrpc/endpoint.pyc", line 113, in consume
  File "pyls_jsonrpc/endpoint.pyc", line 182, in _handle_request
  File "pyls_jsonrpc/dispatchers.pyc", line 23, in handler
  File "/Users/rclary/Documents/Python.git/developer/spyder-mac-app/dist/Spyder.app/Contents/Resources/lib/python3.8/pyls/python_ls.py", line 330, in m_text_document__hover
    return self.hover(textDocument['uri'], position)
  File "/Users/rclary/Documents/Python.git/developer/spyder-mac-app/dist/Spyder.app/Contents/Resources/lib/python3.8/pyls/python_ls.py", line 264, in hover
    return self._hook('pyls_hover', doc_uri, position=position) or {'contents': ''}
  File "/Users/rclary/Documents/Python.git/developer/spyder-mac-app/dist/Spyder.app/Contents/Resources/lib/python3.8/pyls/python_ls.py", line 155, in _hook
    return hook_handlers(config=self.config, workspace=workspace, document=doc, **kwargs)
  File "pluggy/hooks.pyc", line 286, in __call__
  File "pluggy/manager.pyc", line 93, in _hookexec
  File "pluggy/manager.pyc", line 337, in traced_hookexec
  File "pluggy/callers.pyc", line 80, in get_result
  File "pluggy/callers.pyc", line 52, in from_call
  File "pluggy/manager.pyc", line 335, in <lambda>
  File "pluggy/manager.pyc", line 84, in <lambda>
  File "pluggy/callers.pyc", line 208, in _multicall
  File "pluggy/callers.pyc", line 80, in get_result
  File "pluggy/callers.pyc", line 187, in _multicall
  File "/Users/rclary/Documents/Python.git/developer/spyder-mac-app/dist/Spyder.app/Contents/Resources/lib/python3.8/pyls/plugins/hover.py", line 12, in pyls_hover
    definitions = document.jedi_script(position).goto_definitions()
  File "/Users/rclary/Documents/Python.git/developer/spyder-mac-app/dist/Spyder.app/Contents/Resources/lib/python3.8/pyls/workspace.py", line 236, in jedi_script
    sys_path = self.sys_path(environment_path) + extra_paths
  File "/Users/rclary/Documents/Python.git/developer/spyder-mac-app/dist/Spyder.app/Contents/Resources/lib/python3.8/pyls/workspace.py", line 268, in sys_path
    environment = self.get_enviroment(environment_path=environment_path)
  File "/Users/rclary/Documents/Python.git/developer/spyder-mac-app/dist/Spyder.app/Contents/Resources/lib/python3.8/pyls/workspace.py", line 260, in get_enviroment
    environment = jedi.api.environment.create_environment(path=environment_path, safe=False)
  File "/Users/rclary/Documents/Python.git/developer/spyder-mac-app/dist/Spyder.app/Contents/Resources/lib/python3.8/jedi/api/environment.py", line 369, in create_environment
    return Environment(path)
  File "/Users/rclary/Documents/Python.git/developer/spyder-mac-app/dist/Spyder.app/Contents/Resources/lib/python3.8/jedi/api/environment.py", line 67, in __init__
    self._get_subprocess()
  File "/Users/rclary/Documents/Python.git/developer/spyder-mac-app/dist/Spyder.app/Contents/Resources/lib/python3.8/jedi/api/environment.py", line 77, in _get_subprocess
    raise InvalidPythonEnvironment(
jedi.api.environment.InvalidPythonEnvironment: Could not get version information for '/Users/rclary/anaconda3/envs/spy37-15/bin/python3.7': InternalError("The subprocess /Users/rclary/anaconda3/envs/spy37-15/bin/python3.7 has crashed (EOFError('Ran out of input'), stderr=).")
```
</details>

Of particular interest is the line
```
jedi.api.exceptions.InternalError: The subprocess /Users/rclary/anaconda3/envs/spy37-15/bin/python3.7 has crashed (EOFError('Ran out of input'), stderr=)
```
indicating that `jedi` is attempting to perform a subprocess using _the external python console executable_ rather than Spyder's python executable. I don't think that should be happening. This does not seem to occur for bootstrap starts, and I'm not sure why. I believe that `jedi` may be picking up the wrong python executable at startup for the standalone app but not for a bootstrap. Unfortunately, I can't seem to figure out where this is happening as setting pdb traces in the `mainwindow.py` don't seem to work.

@ccordoba12 @goanpeca @CAM-Gerlach,
Do you have any advice here?

**Comment by mrclary** [comment]
Okay, I think I've solved the problem. Should be resolved with PR spyder-ide/spyder#12011

**Comment by ccordoba12** [comment]
Great news!! Sorry for not being able to give you feedback, but I'll try to in the next couple of days.

**Comment by mrclary** [comment]
This issue is fixed with spyder-ide/spyder#12271
